bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/station/NAMainHandler.java

```diff
@@ -107,9 +218,89 @@ protected State getNAThingProperty(String channelId) {
+            case CHANNEL_MIN_HUMIDITY_THIS_WEEK:
+            case CHANNEL_MIN_HUMIDITY_THIS_MONTH:
+            case CHANNEL_MAX_HUMIDITY:
+            case CHANNEL_MAX_HUMIDITY_THIS_WEEK:
+            case CHANNEL_MAX_HUMIDITY_THIS_MONTH:
+                return toQuantityType(channelMeasurements.get(channelId), API_HUMIDITY_UNIT);
+            case CHANNEL_MIN_NOISE:
+            case CHANNEL_MIN_NOISE_THIS_WEEK:
+            case CHANNEL_MIN_NOISE_THIS_MONTH:
+            case CHANNEL_MAX_NOISE:
+            case CHANNEL_MAX_NOISE_THIS_WEEK:
+            case CHANNEL_MAX_NOISE_THIS_MONTH:
+                return toQuantityType(channelMeasurements.get(channelId), API_NOISE_UNIT);
+            case CHANNEL_MIN_PRESSURE:
+            case CHANNEL_MIN_PRESSURE_THIS_WEEK:
+            case CHANNEL_MIN_PRESSURE_THIS_MONTH:
+            case CHANNEL_MAX_PRESSURE:
+            case CHANNEL_MAX_PRESSURE_THIS_WEEK:
+            case CHANNEL_MAX_PRESSURE_THIS_MONTH:
+                return toQuantityType(channelMeasurements.get(channelId), API_PRESSURE_UNIT);
+            case CHANNEL_MIN_TEMP_THIS_WEEK:
+            case CHANNEL_MIN_TEMP_THIS_MONTH:
+            case CHANNEL_MAX_TEMP_THIS_WEEK:
+            case CHANNEL_MAX_TEMP_THIS_MONTH:
+                return toQuantityType(channelMeasurements.get(channelId), API_TEMPERATURE_UNIT);
+            case CHANNEL_DATE_MIN_CO2:
+            case CHANNEL_DATE_MIN_CO2_THIS_WEEK:
+            case CHANNEL_DATE_MIN_CO2_THIS_MONTH:
+            case CHANNEL_DATE_MAX_CO2:
+            case CHANNEL_DATE_MAX_CO2_THIS_WEEK:
+            case CHANNEL_DATE_MAX_CO2_THIS_MONTH:
+            case CHANNEL_DATE_MIN_NOISE:
+            case CHANNEL_DATE_MIN_NOISE_THIS_WEEK:
+            case CHANNEL_DATE_MIN_NOISE_THIS_MONTH:
+            case CHANNEL_DATE_MAX_NOISE:
+            case CHANNEL_DATE_MAX_NOISE_THIS_WEEK:
+            case CHANNEL_DATE_MAX_NOISE_THIS_MONTH:
+            case CHANNEL_DATE_MIN_HUMIDITY:
+            case CHANNEL_DATE_MIN_HUMIDITY_THIS_WEEK:
+            case CHANNEL_DATE_MIN_HUMIDITY_THIS_MONTH:
+            case CHANNEL_DATE_MAX_HUMIDITY:
+            case CHANNEL_DATE_MAX_HUMIDITY_THIS_WEEK:
+            case CHANNEL_DATE_MAX_HUMIDITY_THIS_MONTH:
+            case CHANNEL_DATE_MIN_PRESSURE:
+            case CHANNEL_DATE_MIN_PRESSURE_THIS_WEEK:
+            case CHANNEL_DATE_MIN_PRESSURE_THIS_MONTH:
+            case CHANNEL_DATE_MAX_PRESSURE:
+            case CHANNEL_DATE_MAX_PRESSURE_THIS_WEEK:
+            case CHANNEL_DATE_MAX_PRESSURE_THIS_MONTH:
+            case CHANNEL_DATE_MIN_TEMP_THIS_WEEK:
+            case CHANNEL_DATE_MIN_TEMP_THIS_MONTH:
+            case CHANNEL_DATE_MAX_TEMP_THIS_WEEK:
+            case CHANNEL_DATE_MAX_TEMP_THIS_MONTH:
+                return toDateTimeType(channelMeasurements.get(channelId));
+        }
+
         return super.getNAThingProperty(channelId);
     }
 
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
```

Comment:
Are you sure this is called on handler startup? If it isn't then channelIds will never get populated if channels are already linked when the handler get initialized.

Reply:
Yes, it does